tests/test_viewers.py

```diff
@@ -61,44 +60,175 @@ def test_structure_data_viewer(structure_data_object):
+    assert "Distance" in v.selection_info.value
+    assert "2 atoms selected" in v.selection_info.value
 
     # The x coordinate lower than 0.5.
     v._selected_atoms.value = "x<0.5"
     v.apply_displayed_selection()
     assert v.selection == [0]
+    assert v.displayed_selection == [0]
+    assert "1 atoms selected" in v.selection_info.value
 
     # The id of the second atom
     v._selected_atoms.value = "id > 1"
     v.apply_displayed_selection()
     assert v.selection == [1]
 
     # or of the two selections.
-    v._selected_atoms.value = "x>0.5 or x<0.5"
+    v._selected_atoms.value = "x>=0.5 or x<0.5"
     v.apply_displayed_selection()
     assert v.selection == [0, 1]
 
-    # Check the `_prepare_payload` function used for downloading.
-    format_cases = [
-        (
-            "Extended xyz",
-            """MgpMYXR0aWNlPSIzLjg0NzM3IDAuMCAwLjAgMS45MjM2ODUgMy4zMzE5MiAwLjAgMS45MjM2ODUgMS4xMTA2NCAzLjE0MTM2NCIgUHJvcGVydGllcz1zcGVjaWVzOlM6MTpwb3M6UjozOm1hc3NlczpSOjEgcGJjPSJUIFQgVCIKU2kgICAgICAgMC4wMDAwMDAwMCAgICAgICAwLjAwMDAwMDAwICAgICAgIDAuMDAwMDAwMDAgICAgICAyOC4wODU1MDAwMApTaSAgICAgICAxLjkyMzY4NTAwICAgICAgIDEuMTEwNjQwMDAgICAgICAgMC43ODUzNDEwMCAgICAgIDI4LjA4NTUwMDAwCg==""",
-        ),
-        (
-            "xsf",
-            """Q1JZU1RBTApQUklNVkVDCiAzLjg0NzM3MDAwMDAwMDAwIDAuMDAwMDAwMDAwMDAwMDAgMC4wMDAwMDAwMDAwMDAwMAogMS45MjM2ODUwMDAwMDAwMCAzLjMzMTkyMDAwMDAwMDAwIDAuMDAwMDAwMDAwMDAwMDAKIDEuOTIzNjg1MDAwMDAwMDAgMS4xMTA2NDAwMDAwMDAwMCAzLjE0MTM2NDAwMDAwMDAwClBSSU1DT09SRAogMiAxCiAxNCAgICAgMC4wMDAwMDAwMDAwMDAwMCAgICAgMC4wMDAwMDAwMDAwMDAwMCAgICAgMC4wMDAwMDAwMDAwMDAwMAogMTQgICAgIDEuOTIzNjg1MDAwMDAwMDAgICAgIDEuMTEwNjQwMDAwMDAwMDAgICAgIDAuNzg1MzQxMDAwMDAwMDAK""",
-        ),
-        (
-            "cif",
-            """ZGF0YV9pbWFnZTAKX2NoZW1pY2FsX2Zvcm11bGFfc3RydWN0dXJhbCAgICAgICBTaTIKX2NoZW1pY2FsX2Zvcm11bGFfc3VtICAgICAgICAgICAgICAiU2kyIgpfY2VsbF9sZW5ndGhfYSAgICAgICAzLjg0NzM3Cl9jZWxsX2xlbmd0aF9iICAgICAgIDMuODQ3MzcKX2NlbGxfbGVuZ3RoX2MgICAgICAgMy44NDczNwpfY2VsbF9hbmdsZV9hbHBoYSAgICA2MApfY2VsbF9hbmdsZV9iZXRhICAgICA2MApfY2VsbF9hbmdsZV9nYW1tYSAgICA2MAoKX3NwYWNlX2dyb3VwX25hbWVfSC1NX2FsdCAgICAiUCAxIgpfc3BhY2VfZ3JvdXBfSVRfbnVtYmVyICAgICAgIDEKCmxvb3BfCiAgX3NwYWNlX2dyb3VwX3N5bW9wX29wZXJhdGlvbl94eXoKICAneCwgeSwgeicKCmxvb3BfCiAgX2F0b21fc2l0ZV90eXBlX3N5bWJvbAogIF9hdG9tX3NpdGVfbGFiZWwKICBfYXRvbV9zaXRlX3N5bW1ldHJ5X211bHRpcGxpY2l0eQogIF9hdG9tX3NpdGVfZnJhY3RfeAogIF9hdG9tX3NpdGVfZnJhY3RfeQogIF9hdG9tX3NpdGVfZnJhY3RfegogIF9hdG9tX3NpdGVfb2NjdXBhbmN5CiAgU2kgIFNpMSAgICAgICAxLjAgIDAuMDAwMDAgIDAuMDAwMDAgIDAuMDAwMDAgIDEuMDAwMAogIFNpICBTaTIgICAgICAgMS4wICAwLjI1MDAwICAwLjI1MDAwICAwLjI1MDAwICAxLjAwMDAK""",
-        ),
-    ]
+    # Display 2*2*2 supercell
+    v.supercell = [2, 2, 2]
+    assert len(v.structure) == 2
+    assert len(v.displayed_structure) == 16
 
-    for fmt, out in format_cases:
-        v.file_format.label = fmt
-        assert v._prepare_payload() == out
+    # Test intersection of the selection with the supercell.
```

Comment:
Same here, we should try to have shorter tests for maintainibility.

Reply:
done in https://github.com/aiidalab/aiidalab-widgets-base/pull/373/commits/cb6e6bb83df0f4ebe7086e8e9b69a1e4254691fc.

The `test_structure_data_viewer_selection` remains a bit long, but I think it is fine since it focuses on different ways of atom selection.